src/atomate2/qchem/sets/base.py

```diff
@@ -0,0 +1,495 @@
+                else:
+                    raise RuntimeError(
+                        "nbo params version should only be set to 7! Exiting..."
+                    )
+            nbo_dict = {}
+            for key in self.nbo_params:
+                if key != "version":
+                    nbo_dict[key] = self.nbo_params[key]
+
+        geom_opt_dict = self.new_geom_opt
+        if self.new_geom_opt is not None:
+            rem_dict["geom_opt2"] = "3"
+            if "maxiter" in self.new_geom_opt:
+                if self.new_geom_opt["maxiter"] != str(self.geom_opt_max_cycles):
+                    raise RuntimeError(
+                        "Max # of optimization cycles must be the same! Exiting..."
+                    )
+            else:
+                self.new_geom_opt["maxiter"] = str(self.geom_opt_max_cycles)
+            geom_opt_dict = {}
+            for key in self.new_geom_opt:
+                geom_opt_dict[key] = self.new_geom_opt[key]
+
+        if self.overwrite_inputs:
+            for sub, sub_dict in self.overwrite_inputs.items():
+                if sub == "rem":
+                    temp_rem = lower_and_check_unique(sub_dict)
+                    for k, v in temp_rem.items():
+                        rem_dict[k] = v
+                if sub == "pcm":
+                    temp_pcm = lower_and_check_unique(sub_dict)
+                    for k, v in temp_pcm.items():
+                        pcm_dict[k] = v
+                if sub == "solvent":
+                    temp_solvent = lower_and_check_unique(sub_dict)
+                    for k, v in temp_solvent.items():
+                        solv_dict[k] = v
+                if sub == "smx":
+                    temp_smx = lower_and_check_unique(sub_dict)
+                    for k, v in temp_smx.items():
+                        smx_dict[k] = v
+                if sub == "scan":
+                    temp_scan = lower_and_check_unique(sub_dict)
+                    for k, v in temp_scan.items():
+                        scan_dict[k] = v
+                if sub == "van_der_waals":
+                    temp_vdw = lower_and_check_unique(sub_dict)
+                    for k, v in temp_vdw.items():
+                        vdw_dict[k] = v
+                    # set the PCM section to read custom radii
+                    pcm_dict["radii"] = "read"
+                if sub == "plots":
+                    temp_plots = lower_and_check_unique(sub_dict)
+                    for k, v in temp_plots.items():
+                        plots_dict[k] = v
+                if sub == "nbo":
+                    if nbo_dict is None:
+                        raise RuntimeError(
+                            "Can't overwrite nbo params when NBO is not being run! Exiting..."
+                        )
+                    temp_nbo = lower_and_check_unique(sub_dict)
+                    for k, v in temp_nbo.items():
+                        nbo_dict[k] = v
```

Comment:
could be replaced by something like:

```python
prop_dicts = {
    "rem": rem_dict,
    ...
    "plots": temp_plots,
}
if self.overwrite_inputs:
    for sub, sub_dict in self.overwrite_inputs.items():
        prop_dict = prop_dicts.get(sub, {})
        for k, v in lower_and_check_unique(sub_dict):
            prop_dict[k] = v
```

Reply:
Keeping this unchanged for now,

@orionarcher , I feel that this doesn't change the current implementation too much. Agreed that it's long-winded, but the new `prop_dicts` will also be clunky.